broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -697,7 +703,19 @@ public void onFailure() {
 
   @Override
   public void onRecovered() {
-    actor.run(() -> updateHealthStatus(HealthStatus.HEALTHY));
+    actor.run(this::onRecoveredInternal);
+  }
+
+  private void onInstallFailure() {
+    updateHealthStatus(HealthStatus.UNHEALTHY);
+    if (atomixRaftPartition.getRole() == Role.LEADER) {
+      LOG.info("Unexpected failures occurred when installing leader services, stepping down");
+      atomixRaftPartition.stepDown();
```

Comment:
If we would step down on failure we would try to reinstallt the follower partition right ? Maybe this makes also sense?

Reply:
Did you mean we step down on all failures? Please see my other comment.